### PR TITLE
prometheus: Move the prometheus nil check inside the prometheus function

### PR DESCRIPTION
**Pull request by roidelapluie**
That pattern will be reused in future metrics.

Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

### CODE REVIEW

**Comment by purpleidea** [comment]
LGTM assuming tests pass.

**Comment by purpleidea** [comment]
FAIL:
> prometheus/prometheus.go:84: not enough arguments to return

File: prometheus/prometheus.go [hunk 1]
```diff
@@ -80,6 +80,9 @@ func (obj *Prometheus) Stop() error {
 // UpdateCheckApplyTotal refreshes the failing gauge by parsing the internal
 // state map.
 func (obj *Prometheus) UpdateCheckApplyTotal(kind string, apply, eventful, errorful bool) error {
+	if obj == nil { // when mgmt is launched without --prometheus
```
Comment (purpleidea) [inline prometheus/prometheus.go]: return XXX and you can do this all on one line too :)

**Comment by roidelapluie** [comment]
On 25 Feb 23:54, James wrote:
> purpleidea commented on this pull request.
> 
> 
> 
> > @@ -80,6 +80,9 @@ func (obj *Prometheus) Stop() error {
>  // UpdateCheckApplyTotal refreshes the failing gauge by parsing the internal
>  // state map.
>  func (obj *Prometheus) UpdateCheckApplyTotal(kind string, apply, eventful, errorful bool) error {
> +	if obj == nil { // when mgmt is launched without --prometheus
> 
> return XXX and you can do this all on one line too :)


done

> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/purpleidea/mgmt/pull/167#pullrequestreview-23875818

-- 
 (o-    Julien Pivotto
 //\    Open-Source Consultant
 V_/_   Inuits - https://www.inuits.eu

**Comment by roidelapluie** [comment]
On 25 Feb 23:54, James wrote:
> FAIL:
> > prometheus/prometheus.go:84: not enough arguments to return

fixed

> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/purpleidea/mgmt/pull/167#issuecomment-282539660

-- 
 (o-    Julien Pivotto
 //\    Open-Source Consultant
 V_/_   Inuits - https://www.inuits.eu

**Pull request merged**